src/khoj/interface/web/chat.html

```diff
@@ -183,12 +183,18 @@
                         referenceSection.appendChild(polishedReference);
                     }
                 }
-        }
+            }
 
             return numOnlineReferences;
-                    }
+        }
+
+        function renderMessageWithReference(message, by, context=null, dt=null, onlineContext=null, intentType=null) {
+            if (intentType === "text-to-image") {
```

Comment:
Great point! Maybe the API response should be streamed with the image rendered as markdown. This would avoid each of the clients rendering the markdown themselves. Let's just do it once we actually get to combining text and image in single response